### PR TITLE
[generate format] ensure 100% coverage of Libra transactions

### PR DESCRIPTION
**Pull request by matbd**
## Summary

* Add a simple proptest strategy to cover multisig authenticators
* Some nits in crypto conversion code to follow Rust best practices
* Improve error messages in Serde Reflection (used to debug the next issue)
* Make sure that Keys, EventKeys, Hashes are deserialized as Vec<u8> instead of Bytes.
* Turn on deserialization tracing by default.

In the future, we may also switch both serialization and deserialization of Keys, EventKeys, Hashes to Bytes (inducing a visible format change in python clients, though). Pending future improvement of Serde-reflection's deserializer, this would allow us to spare one data copy at deserialization time.

## Test Plan

```
cargo x test -p generate-format
```

### CODE REVIEW

File: types/src/transaction/mod.rs [hunk 1]
```diff
@@ -235,6 +235,17 @@ impl RawTransaction {
         )))
     }
 
+    pub fn multi_sign_for_testing(
```
Comment (davidiw) [inline types/src/transaction/mod.rs:238]: shouldn't this be exposed only for 

#[cfg(any(test, feature = "fuzzing"))]

or maybe even inline where you are using it?

Comment (matbd) [reply types/src/transaction/mod.rs:239]: Yes, thanks. Good idea!

**Comment by thefallentree** [comment]
How do we see the actual coverage? is there an test that can detect if the coverage is less than 100% in the futrue?

**Comment by matbd** [comment]
@thefallentree After the last commit "switch on deserialization tracing by default", the existing unit test will break if it fails to guarantee 100% coverage (for types under `Transaction`).

In a next PR, I am going to improve error message(s) that can happen when deserialization-tracing fails. Hopefully, this will make the whole tracing process more transparent.

**Pull request closed without merging**